extensions/widgets/segmented/tests/highlight.livecodescript

```diff
@@ -0,0 +1,165 @@
+
+	set the hilitedItemNames of sWidget to empty
+	TestAssert "can clear hilitedItemNames", \
+			the hilitedItems of sWidget is "" and \
+			the number of items in the hilitedItemNames of sWidget is 0 and \
+			the hilitedItemNames of sWidget is empty
+
+	---------- Multiple highlight
+	set the multipleHilites of sWidget to true
+	set the hilitedItemNames of sWidget to ",segment1"
+	TestAssert "order in hilitedItemNames doesn't matter", \
+			the hilitedItems of sWidget is "1,4" and \
+			the number of items in the hilitedItemNames of sWidget is 2 and \
+			item 1 to 2 of the hilitedItemNames of sWidget is item 1 to 2 of "segment1,,,"
+
+	set the hilitedItemNames of sWidget to "segment1,segment2,segment4"
+	TestAssert "unknown segments in hilitedItemNames are ignored", \
+			the hilitedItems of sWidget is "1,2" and \
+			the number of items in the hilitedItemNames of sWidget is 2 and \
+			(item 1 to 2 of the hilitedItemNames of sWidget is \
+					item 1 to 2 of "segment1,segment2")
+
+	set the hilitedItemNames of sWidget to "segment1,segment2,segment1"
+	TestAssert "duplicate segments in hilitedItemNames are ignored", \
+			the hilitedItems of sWidget is "1,2" and \
+			the number of items in the hilitedItemNames of sWidget is 2 and \
+			(item 1 to 2 of the hilitedItemNames of sWidget is \
+					item 1 to 2 of "segment1,segment2")
+
+	---------- Duplicate names
+	-- These tests are intended to test the pathological case where
+	-- there are duplicate itemNames
+	set the itemNames of sWidget to "A,A,B,B"
+	set the hilitedItems of sWidget to "2,4"
+	TestAssert "hilitedItemNames reported correctly when names duplicated", \
+			the hilitedItems of sWidget is "2,4" and \
+			the hilitedItemNames of sWidget is "A,B"
+
+	set the hilitedItemNames of sWidget to "B,A"
+	TestAssert "setting hilitedItemNames takes first duplicated name", \
+			the hilitedItems of sWidget is "1,3" and \
+			the hilitedItemNames of sWidget is "A,B"
+end TestHilitedItemNames
+
+on TestMultipleHilites
+	TestAssert "single highlight by default", \
+			the multipleHilites of sWidget is false
+
+	set the multipleHilites of sWidget to true
+	set the hilitedItems of sWidget to "1,2,3"
+
+	put false into sHiliteChanged
+	set the multipleHilites of sWidget to false
+	wait 0 with messages
+	TestAssert "setting single highlight resets multiple highlight", \
+			the hilitedItems of sWidget is empty
+	TestAssert "resetting highlight sends hiliteChanged", \
+			sHiliteChanged
+
+	set the multipleHilites of sWidget to true
+	set the hilitedItems of sWidget to "2"
+	put false into sHiliteChanged
+	set the multipleHilites of sWidget to false
+	wait 0 with messages
+	TestAssert "setting single highlight preserves single highlight", \
+			the hilitedItems of sWidget is "2"
+	TestAssert "preserving single highlight doesn't send hiliteChanged", \
+			sHiliteChanged
+end TestMultipleHilites
+
+-- Local Variables:
+-- tab-width: 3
+-- indent-tabs-mode: t
+-- End:
```

Comment:
@peter-b if you indent livecode script with tabs then every time someone touches it with the IDE editor it will cause whitespace diff headaches.